src/TLuaInterpreter.cpp

```diff
@@ -13769,3 +13770,21 @@ int TLuaInterpreter::startPermSubstringTrigger(const QString & name, const QStri
 
 }
 
+int TLuaInterpreter::alert( lua_State * L )
+{
+    int luaDurationMs;
+    if( ! lua_isnumber( L, 1 ) )
+    {
+        lua_pushstring( L, "alert: wrong argument type" );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaDurationMs = lua_tointeger( L, 1 );
+    }
+
+    QApplication::alert(mudlet::self(), luaDurationMs);
```

Comment:
The Qt Documentation for this method is:
> `[static] void QApplication::alert(QWidget *widget, int msec = 0)`
> Causes an alert to be shown for widget if the window is not the active window. The alert is shown for `msec` miliseconds. If `msec` is zero (the default), then the alert is shown indefinitely until the window becomes active again.
> Currently this function does nothing on Qt for Embedded Linux.
> On macOS, this works more at the application level and will cause the application icon to bounce in the dock.
> On Windows, this causes the window's taskbar entry to flash for a time. If `msec` is zero, the flashing will stop and the taskbar entry will turn a different color (currently orange).
> On X11, this will cause the window to be marked as "demands attention", the window must not be hidden (i.e. not have `hide()` called on it, but be visible in some sort of way) in order for this to work.
>This function was introduced in Qt 4.3.

So, the behaviour does have some platform variations! 😮  I think there ought to be some validation of the value being 0 or greater - which being a _run-time_ error should (according to what @vadi2 has recently drilled into me) be something along the lines of:

    duration, in miliiseconds, must be zero or greater.

or if we make it optional:

    duration, in milliseconds, is optional but if given must be zero or more.